### ISSUE TITLE
Next Release (planning)

### ISSUE THREAD

**Issue by machawk1** [oduwsdl/ipwb#644]
Let's prime for another release of ipwb!

We are currently waiting on an updated release of ipfshttpclient (py module) before doing so (see https://github.com/ipfs-shipyard/py-ipfs-http-client/issues/204) to mitigate the need to reference their master branch (#638) due to some lagging with the latest version of go-ipfs (v0.5.0). We have some other GitHub issues that have been closed (thanks to @anatoly-scherbakov), some unreleased code that has been merged to master, and other contributions that need some further review before being merged.

**Target**: May 11, 2020

**Need to review, revise, and merge**: #631/#604, #640

**Other tasks**:
- [ ] When ipfshttpclient pushes another release, change ipwb to again reference a release version instead of the master branch.
- [ ] Further tickets to target completion? (/cc @ibnesayeed)

**Comment by machawk1** [comment]
It is now May 11, 2020 and no new release from py-ipfs-http-client. I would like to move forward with pushing a release based on an unreleased version of the module as a dependency either from their master or pinned to the latest commit on their master. Any objections, @ibnesayeed?

**Comment by ibnesayeed** [comment]
I don;t have any objections other than the concerns I have already registered earlier in another ticket. That said, if IPFS 0.5 support is critical now then please go ahead, but if we can wait for a few more days, we should!

**Comment by machawk1** [comment]
@ibnesayeed I believe IPFS 0.5 support is now critical. Let's release then perhaps push another with the above contributions resolved and merged in.

**Comment by ibnesayeed** [comment]
Then go for it with the `master` branch, pinning to a commit hash might have other issues of not being able to fetch it. If we can afford, we may want to release a `beta` or `rc` version so people who desperately need IPFS 0.5 support can download it, but otherwise know that this version will eventually be erased later with a more stable and reproducible release.

**Comment by ibnesayeed** [comment]
@github-actions run

```sh
#! /bin/sh

exec > comment.buffer

echo "## Changes Since Last Release"

echo "### Commits"
git log $(git describe --tags --abbrev=0)..@ --oneline | grep -v "Merge pull request" | awk '{print "*",$0}'

echo "### Changed Files"
echo "\`\`\`"
git diff --stat $(git describe --tags --abbrev=0)..@
echo "\`\`\`"
```

**Comment by ibnesayeed** [comment]
@github-actions run

```sh
#! /bin/sh

exec > comment.buffer

LASTTAG=$(git describe --tags --abbrev=0)

echo "## Changes Since Last Release"

echo "### Commits"
git log $LASTTAG..@ --oneline | grep -v "Merge pull request" | awk '{print "*",$0}'

echo "### Changed Files"
echo "\`\`\`"
git diff --stat $LASTTAG..@
echo "\`\`\`"

echo "### Contributors"
git log --format="%an: %s" $LASTTAG..@ | awk -F":" '{a[$1]++}END{for(k in a){print "* "k" ("a[k]" commits)"}}'
```

**Comment by ibnesayeed** [comment]
@github-actions run

```sh
#! /bin/sh

exec > comment.buffer

LASTTAG=$(git describe --tags --abbrev=0)

echo "## Changes Since Last Release"

echo "### Pull Requests"
git log --format="%s %b" $LASTTAG..@ | grep "Merge pull request" | cut -d' ' -f4,7- | awk '{print "* "$0}'

echo "### Commits"
git log $LASTTAG..@ --oneline | grep -v "Merge pull request" | awk '{print "* "$0}'

echo "### Changed Files"
echo "\`\`\`"
git diff --stat $LASTTAG..@
echo "\`\`\`"

echo "### Contributors"
git log --format="%an: %s" $LASTTAG..@ | grep -v "Merge pull request" | awk -F":" '{a[$1]++}END{for(k in a){print "* "k" ("a[k]" commits)"}}'
```

**Comment by ibnesayeed** [comment]
@github-actions run

```sh
#! /bin/sh

exec > comment.buffer

LASTTAG=$(git describe --tags --abbrev=0)

echo "## Changes Since Last Release"

echo "### Pull Requests"
git log --format="%s %b" $LASTTAG..@ | grep "Merge pull request" | cut -d' ' -f4,7- | awk '{print "* "$0}'

echo "### Commits"
git log $LASTTAG..@ --oneline | grep -v "Merge pull request" | awk '{print "* "$0}'

echo "### Changed Files"
echo "\`\`\`"
git diff --stat $LASTTAG..@
echo "\`\`\`"

echo "### Contributors"
git log --format="%an: %s" $LASTTAG..@ | grep -v "Merge pull request" | awk -F":" '{a[$1]++}END{for(k in a){print "* "k" ("a[k]" commits)"}}'
```

**Comment by ibnesayeed** [comment]
@github-actions run

```sh
#! /bin/sh

exec > comment.buffer

LASTTAG=$(git describe --tags --abbrev=0)

echo "## Changes Since Last Release"

echo "### Pull Requests"
git log --format="%s %b" $LASTTAG..@ | grep "Merge pull request" | cut -d' ' -f4,7- | awk '{print "* "$0}'

echo "### Commits"
git log $LASTTAG..@ --oneline | grep -v "Merge pull request" | awk '{print "* "$0}'

echo "### Changed Files"
echo "\`\`\`"
git diff --stat $LASTTAG..@
echo "\`\`\`"

echo "### Contributors"
git log --format="%an: %s" $LASTTAG..@ | grep -v "Merge pull request" | cut -d":" -f1 | tr ' ' '#' | sort | uniq -c | sort -nr | awk '{print "* "$2" ("$1" commits)"}' | tr '#' ' '
```

**Comment by machawk1** [comment]
The official ipfshttpclient module has yet to push another release to Pypi. I created https://pypi.org/project/ipfshttpclient4ipwb/ based on the current master, as Pypi won't allow direct references. The ipwb references to this dependency need to be updated from ipfshttpclient.

**Comment by ibnesayeed** [comment]
@github-actions run

```sh
#! /bin/sh

exec > comment.buffer

LASTTAG=v0.2020.04.24.1847

echo "## Changes Since Last Release"

echo "### Pull Requests"
git log --format="%s %b" $LASTTAG..@ | grep "Merge pull request" | cut -d' ' -f4,7- | awk '{print "* "$0}'

echo "### Commits"
git log $LASTTAG..@ --oneline | grep -v "Merge pull request" | awk '{print "* "$0}'

echo "### Changed Files"
echo "\`\`\`"
git diff --stat $LASTTAG..@
echo "\`\`\`"

echo "### Contributors"
git log --format="%an: %s" $LASTTAG..@ | grep -v "Merge pull request" | cut -d":" -f1 | tr ' ' '#' | sort | uniq -c | sort -nr | awk '{print "* "$2" ("$1" commits)"}' | tr '#' ' '
```

**Comment by machawk1** [comment]
Closing this issue, @ibnesayeed, as we mitigated the ipfshttpclient for now. We can make another issue or use a different means (e.g., GH Projects) to track the next release. I am hoping we can get the binsearch (#604) work integrated.

**Comment by ibnesayeed** [comment]
We need to streamline the release process with automated release notes, but we can track that separately.

**Comment by machawk1** [comment]
@ibnesayeed You are a little more familiar with that auto-generation process (which I would like to see documented), so please create a GitHub issue of what you anticipate being the ideal process and we can work toward that.

**Comment by ibnesayeed** [comment]
I am actually waiting for some changes in an upstream GH action (and have provided feedback there). Depending on how things go, I can create workflow files to achieve the goal. Also, I perhaps will create another GH reusable action to aid the release notes process.

**Comment by ibnesayeed** [comment]
@github-actions run

```sh
#! /bin/bash

exec > comment.buffer

git log --format="%(trailers:key=Co-authored-by,valueonly)" v0.2020.04.24.1847..@ | awk -F' <' '{print $1}'
```
